nemo/nemo/core/neural_modules.py

```diff
@@ -115,15 +124,19 @@ def __call__(self, **kwargs):
         #    raise ValueError("We currently do not support calling same NM"
         #                     "more than once")
 
+        # Get input and output ports definitions.
+        input_port_defs = self.input_port_definitions()
+        output_port_defs = self.output_port_definitions()
+
         first_input_nmtensor_type = None
         input_nmtensors_are_of_same_type = True
         for port_name, tgv in kwargs.items():
-            if port_name not in self._input_ports.keys():
+            if port_name not in input_port_defs.keys():
```

Comment:
basically, if you address my comment above, changes like these are unnecessary

Reply:
I find those names more appropriate - we are talking about _port definitions_ not _ports_ per se.

Reply:
Keeping old names keeps more backward of the compatibility intact and is less error prone.
And if we want to debate over the name (which I don't think we should because backwards compatibility precedes that) I find both "input_port_definitions" and "input_ports" too long, should be just changed to "inputs"

Reply:
1) we are discussing changing of *two lines of code inside of neural_module*
2) I guess we both agree on naming convention that *underscore* at the beginning shouldn't be used for names of externally visible methods/properties. So the name _input_ports cannot remain as it is, am I correct?